service-userlayer/src/main/java/org/oskari/map/userlayer/input/FeatureCollectionParsers.java

```diff
@@ -0,0 +1,86 @@
+package org.oskari.map.userlayer.input;
+
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.data.simple.SimpleFeatureIterator;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.geotools.feature.DefaultFeatureCollection;
+import org.geotools.geometry.jts.JTS;
+import org.geotools.referencing.CRS;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.operation.MathTransform;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+public class FeatureCollectionParsers {
+
+    private FeatureCollectionParsers() {}
+
+    public static boolean hasByFileExt(String fileExt) {
+        if (fileExt == null) {
+            return false;
+        }
+        fileExt = fileExt.toUpperCase();
+        switch (fileExt) {
+        case GPXParser.SUFFIX:
+        case KMLParser.SUFFIX:
+        case MIFParser.SUFFIX:
+        case SHPParser.SUFFIX:
+            return true;
+        default:
+            return false;
+        }
+    }
+
+    public static FeatureCollectionParser getByFileExt(String fileExt) {
+        if (fileExt == null) {
+            return null;
+        }
+        fileExt = fileExt.toUpperCase();
+        switch (fileExt) {
+        case GPXParser.SUFFIX: return new GPXParser();
+        case KMLParser.SUFFIX: return new KMLParser();
+        case MIFParser.SUFFIX: return new MIFParser();
+        case SHPParser.SUFFIX: return new SHPParser();
+        default: return null;
+        }
+    }
+
+    /**
+     * Read Features from FeatureSource to memory (ensure they won't
+     * disappear when DataStore#dispose() is called) while transforming
+     * their geometries from source projection to target projection
+     * @throws Exception lots can go wrong
+     */
+    public static SimpleFeatureCollection read(SimpleFeatureSource src,
+            CoordinateReferenceSystem sourceCRS, CoordinateReferenceSystem targetCRS) throws Exception {
+        DefaultFeatureCollection fc = new DefaultFeatureCollection();
+        SimpleFeatureCollection sfc = src.getFeatures();
+        MathTransform transform = getTransform(sourceCRS, targetCRS);
```

Comment:
Does transform need a null check here?

Reply:
If either sourceCRS or targetCRS is null getTransform will return null. The `transform` is scoped to this function, and it's null-ness is checked on line 64.

Reply:
Changed the way this works. Now if either sourceCRS or targetCRS is null we will throw an exception instead.